### ISSUE TITLE
Support for definitions via the `setup` function

### ISSUE THREAD

**Issue by kkelk** [rlaffers/eslint-plugin-xstate#28]
The xstate documentation's recommended approach for specifying types in Typescript is [`to use the setup function`](https://stately.ai/docs/typescript#specifying-types). This then chains into a normal `createMachine` call, but this pattern is not currently detected by `eslint-plugin-xstate`.

**Comment by rlaffers** [comment]
Hi @kkelk 
what is your specific code where you are getting an ESLint error, or else hope to get an ESLint error?

**Comment by junwen-k** [comment]
I agree with this suggestion, and I believe it's important to update the plugin to support this pattern. With the current implementation, the following code is linted correctly:

```ts
const machine = createMachine({ ... });
```

However, this code, which follows the recommended approach using `setup`, is not linted:

```ts
const machine = setup({ ... }).createMachine({ ... });
```

I am guessing that this [file](https://github.com/rlaffers/eslint-plugin-xstate/blob/master/lib/utils/predicates.js) needs to be updated.

To ensure compatibility with the latest XState best practices, it would be great to add support for this pattern in the plugin. Thanks for considering this update!
